admin-api-server/src/analytics/service/analytics.service.ts

```diff
@@ -258,38 +285,58 @@ LIMIT ${params.pageSize}
     }
 
     return {
-      data: qryRes.rows.map((row: any) => {
-        const price = this.currencyService.convertToCurrency(
-          Number(row['price']),
-          BASE_CURRENCY,
-        );
+      data: await Promise.all(
+        qryRes.rows.map(async (row: any) => {
+          const price = this.currencyService.convertToCurrency(
+            Number(row['price']),
+            BASE_CURRENCY,
+          );
+          const historicRates = await this.currencyService.ratesAt(
+            row['timestamp'],
+          );
+          const feeInBaseCurrency: number =
+            this.currencyService.convertFromCurrency(
+              Number(row['mutez_fee']) / 1000000,
+              'XTZ',
+              historicRates,
+              false,
+            );
+          const nftPriceSumBase = this.currencyService.convertToCurrency(
+            row['nft_price_sum'],
+            BASE_CURRENCY,
+          );
+          const conversionRate =
+            row['nft_order_price'] &&
+            Number(nftPriceSumBase) / row['nft_order_price'];
 
-        const nftPriceSumBase = this.currencyService.convertToCurrency(
-          row['nft_price_sum'],
-          BASE_CURRENCY,
-        );
-        const conversionRate =
-          row['nft_order_price'] &&
-          Number(nftPriceSumBase) / row['nft_order_price'];
-
-        return <Activity>{
-          id: Number(row['id']),
-          timestamp: Math.floor(row['timestamp'].getTime() / 1000),
-          kind: row['kind'],
-          from: row['from'],
-          to: row['to'],
-          tokenId: Number(row['tokenId']),
-          price,
-          edition_size: Number(row['edition_size']),
-          currency: row['currency'],
-          transaction_value:
-            price &&
-            conversionRate &&
-            (Number(price) / conversionRate).toFixed(2),
-          conversion_rate: conversionRate && conversionRate.toFixed(2),
-        };
-      }),
-      count: Number(qryRes.rows[0]['total_activity_count']),
+          return <Activity>{
+            id: Number(row['id']),
+            timestamp: Math.floor(row['timestamp'].getTime() / 1000),
+            kind: row['kind'],
+            from: row['from'],
+            to: row['to'],
+            tokenId: Number(row['tokenId']),
+            price,
+            edition_size: Number(row['edition_size']),
+            currency: row['currency'],
+            transaction_value:
+              price &&
+              conversionRate &&
+              (Number(price) / conversionRate).toFixed(2),
+            conversion_rate: conversionRate && conversionRate.toFixed(2),
+            purchaser_country: row['purchaser_country'],
+            fee_in_base_currency:
+              feeInBaseCurrency > 0
+                ? Number(
+                    this.currencyService
+                      .convertFromBaseUnit(BASE_CURRENCY, feeInBaseCurrency)
```

Comment:
There is a `toFixedDecimals` in the currencyService for this. This will apply the correct number of fixed decimal points (you applied 4 below, but this is 2 too much for fiat currencies, and if the base currency is Tez then 4 is wrong as well because then it should be 6 decimal points).

Reply:
Removed it. It can now happen that the number becomes very long e.g. 0.0006013500000000001€

Reply:
ooh I remember again, sorry I had forgotten that these numbers are suuper small and can easily be fractions of a cent.

In that case yes, the way it is now (without truncation) is for now the best thing to do. Lets wait for feedback before deciding how to truncate.